test/jdk/java/awt/dnd/DragToAnotherScreenTest.java

```diff
@@ -0,0 +1,169 @@
+                                               .getName());
+
+        GraphicsDevice[] gd = GraphicsEnvironment.getLocalGraphicsEnvironment()
+                                                 .getScreenDevices();
+        if (gd.length == 1) {
+            PassFailJFrame.log("Single Monitor");
+        } else {
+            PassFailJFrame.log("Multi-Monitor");
+        }
+        PassFailJFrame.log("--------------");
+        PassFailJFrame.log("Test logs:\n");
+        Frame frame0 = new Frame("Drag Source", gd[0].getDefaultConfiguration());
+        frame0.setSize(300, 300);
+        label0 = new Label("Drag me");
+        frame0.add(label0);
+
+        Frame frame1 = new Frame("Drop Target", gd[(gd.length > 1 ? 1 : 0)].getDefaultConfiguration());
+        frame1.setSize(300, 300);
+        label1 = new Label("Drop on me");
+        frame1.add(label1);
+
+        DragGestureListener dragGestureListener = dge -> dge.startDrag(null, new StringSelection(label0.getText()), null);
+        new DragSource().createDefaultDragGestureRecognizer(label0,
+                                                            DnDConstants.ACTION_COPY, dragGestureListener);
+
+        DropTargetAdapter dropTargetAdapter = new DropTargetAdapter() {
+            public void drop(DropTargetDropEvent dtde) {
+                Transferable t = dtde.getTransferable();
+                if (t.isDataFlavorSupported(DataFlavor.stringFlavor)) {
+                    dtde.acceptDrop(DnDConstants.ACTION_COPY);
+                    try {
+                        String str = (String) t.getTransferData(DataFlavor.stringFlavor);
+                        label1.setText(str);
+                        JOptionPane.showMessageDialog(frame0,
+                                                      "getTransferData was successful",
+                                                      "Test Passed", JOptionPane.PLAIN_MESSAGE);
+                        PassFailJFrame.forcePass();
+                    } catch (Exception e) {
+                        dtde.dropComplete(false);
+                        PassFailJFrame.log("getTransferData() Failed: " + e);
+                        JOptionPane.showMessageDialog(frame0,
+                                                      "getTransferData() Failed",
+                                                      "Test Failed", JOptionPane.ERROR_MESSAGE);
+                        PassFailJFrame.forceFail();
+                    }
+                    dtde.dropComplete(true);
+                } else {
+                    dtde.rejectDrop();
+                    PassFailJFrame.log("stringFlavor is not supported by Transferable");
+                    JOptionPane.showMessageDialog(frame0,
+                                                  "stringFlavor is not supported by Transferable",
+                                                  "Test Failed", JOptionPane.ERROR_MESSAGE);
+                    PassFailJFrame.forceFail();
```

Comment:
Provide the reason:
```suggestion
                    PassFailJFrame.forceFail("stringFlavor is not supported by Transferable");
```

Reply:
Added it , additionally logging the errors in PFJ.logArea()